client/my-sites/plans-features-main/index.jsx

```diff
@@ -79,11 +87,26 @@ export class PlansFeaturesMain extends Component {
 			selectedPlan,
 			withDiscount,
 			siteId,
+			siteType,
 			plansWithScroll,
 		} = this.props;
 
 		const plans = this.getPlansForPlanFeatures();
 		const visiblePlans = this.getVisiblePlansForPlanFeatures( plans );
+
+		const popularPlanSpec =
+			! siteType || abtest( 'popularPlanBy' ) === 'customerType'
+				? {
+						// Control experience
+						type: customerType === 'personal' ? TYPE_PREMIUM : TYPE_BUSINESS,
+						group: GROUP_WPCOM,
+				  }
+				: {
+						// Testing suggesting plans by siteType
+						type: getPopularPlanType( siteType ),
+						group: GROUP_WPCOM,
+				  };
+
```

Comment:
The `PlansFeaturesMain` component is a weird mix of presentation and business logic jumbled together, growing in complexity as we develop even more permutations of signup flows that affect how we display plans.

I believe this component displays for both signup flows and /plans/{site}. Is this a/b test meant to affect `/plans/{site}`, as well? If not, I wonder if what's happening here to determine the popular plan is better placed in the `PlansStep` and passed to this component, so that `popularPlanSpec` here becomes a simple prop that defaults to the Business plan.

Reply:
Another related factor: how is this mean to effect the display of Jetpack plans?

Reply:
Thanks for taking a look, @creativecoder.  You're right about the logic being messy in this component.  I went ahead and pulled it out into a helper function in the library so it's simpler.  I tried it out in the `PlansStep`, but that didn't seem quite right either.

I explicitly handled Jetpack sites as well.

> I believe this component displays for both signup flows and /plans/{site}. Is this a/b test meant to affect /plans/{site}, as well?

That's correct. `PlansFeaturesMain` is used both in signup and inside calypso (`/plans/{site}`). Personally, I think the abtest should apply both places... It doesn't make a lot of sense to show someone that one plan is popular during signup and change that statement once they've signed up and are looking to upgrade -- at least without some other input from them purposefully changing our recommendation.

If we _do_ want to vary the behavior, we can leverage the [`isInSignup` prop](https://github.com/Automattic/wp-calypso/blob/ff5d68e573bf791c643c27c787d7b4e4bd83fd19/client/my-sites/plans-features-main/index.jsx#L369).